frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -806,12 +833,25 @@ func (dss *dispatchStates) addState(ds *dispatchState) {
 	if d, ok := dss.statesByName[ds.stage.BaseName]; ok {
 		ds.base = d
 		ds.outline = d.outline.clone()
+		// Propagate the set of paths from this dispatchState to the base
+		// dispatchState to enable filtering local contexts based on the set
+		// of used paths.
+		ds.paths = mergePaths(ds.paths, d.paths)
```

Comment:
isn't it a problem that this runs before `dispatchRun`.

I'm not sure how a path added in the dispatchRun of a child stage would affect the dispatchState of the parent that has `namedContext` set.

Reply:
I think I should probably change the name of this. I didn't think of a better name yesterday.

The intention of this method is to ensure that `ds.paths` and `d.paths` refer to the same location in memory. At this point, `ds.paths` is probably empty but it's just making sure that everything is present in case it wasn't. This happens before `dispatchRun` because it ensures that all of the maps relate to the same map.

I think `unifyPaths` is probably a better name now that I had a night to sleep on it.